roles/platform/tasks/initialize_setup_aws.yml

```diff
@@ -47,35 +47,42 @@
     plat__aws_xaccount_external_id: "{{ plat__cdp_xaccount_external_id }}"
     plat__aws_xaccount_account_id: "{{ plat__cdp_xaccount_account_id }}"
 
-# TODO - Confirm the two following tasks are the design pattern we want: checking for a set_fact from another role before establishing its own role fact
-- name: Discover AWS VPC
-  when: infra__aws_vpc_id is undefined
+# Runlevel first, upstream second, and discover third
+- name: Discover AWS VPC if not defined
+  when: plat__aws_vpc_id == "" and infra__aws_vpc_id is undefined
   block:
-    - name: Query AWS VPC
+    - name: Query AWS VPC by name
       amazon.aws.ec2_vpc_net_info:
         region: "{{ plat__region }}"
         filters:
           "tag:Name": "{{ plat__vpc_name }}"
       register: __aws_vpc_info
 
+    - name: Assert discovered AWS VPC
+      ansible.builtin.assert:
+        that: __aws_vpc_info.vpc | length == 1
```

Comment:
Hey @wmudge - I think this should be `__aws_vpc_info.vpcs` (i.e. adding the `s` to vpc). Docs on this module are at https://docs.ansible.com/ansible/2.10/collections/amazon/aws/ec2_vpc_net_info_module.html

Reply:
yes, it should

Reply:
Fixed in `runtime` too